include/xtt/context.h

```diff
@@ -187,6 +188,17 @@ struct xtt_client_handshake_context {
     union {
         xtt_ecdsap256_priv_key ecdsap256;
     } longterm_private_key;
+
+#ifdef USE_TPM
```

Comment:
I don't love the complexity of shoving more stuff into the main context here.  Refactoring as with the `crypto/{aead,hmac,kx}.h` functions is needed.

But the pragmatic thing is probably to merge this is as is, taking up the refactoring as a separate PR.

Reply:
Here's a sketch of what the refactoring might look like: https://github.com/xaptum/xtt/commit/e761538a330147f212235e09eaa6af70df16cab3

Reply:
> But the pragmatic thing is probably to merge this is as is, taking up the refactoring as a separate PR

That's my strong preference

I agree that doing it this way is sub-optimal. But, as I discussed in the linked issue, this project needs significant refactoring and it doesn't seem like a good idea to block this new feature in order to start cleaning things up.